lib/core.js

```diff
@@ -0,0 +1,38 @@
+const helpers = require("./helpers");
+const consts = require("./consts.json");
+
+function getServiceInstance(
+  AWSService,
+  actionParams,
+  settings,
+  credentialsLabels = consts.DEFAULT_CREDENTIAL_LABELS,
+) {
+  this.credentials = helpers.readCredentials(actionParams, settings, credentialsLabels);
+  return new AWSService(this.credentials);
+}
+
+function handleInput(awsService, action, settings, overrideParsers = {}) {
```

Comment:
This is just some random function, returning random data and functionalities used in some combination later on. That not only isn't `input`, but also `handle` is too vague. Is there no way to split it and name accordingly?

Reply:
I don't agree. This function does a lot of stuff that needs to happen (almost) every time at the start of the method code. It creates the aws service instance with credentials injected, parses all the parameters passed  to the method and returns those prepared data back to the developer, so that he can use this to implement the logic and not bother with technical details. I can agree to changing the name if you think `handle` is too vague, however action and settings are literally the input to the method and this function handles all of the technical mambo-jumbo the developer shouldn't have to care about – I don't really see anything random about that. 

I think the main misunderstanding here might come from the perspective you look at this function. If you only see it as something `mapToAwsMethod` uses, then true – this is random and probably shouldn't exist in such form, especially that this is an aggregation function which pretty much composes 3 other function calls. But this is not the intent for this function at all. This is a QOL function for the developers to use as the first thing when implementing the method logic (please see EC2 or S3 implementations for context) and as that I believe this function is very important and useful. 

Also maybe it would be easier to see from that perspective if the library's module structure was a little refined, so that the API which is intended to be used directly by developers was more obviously highlighted?